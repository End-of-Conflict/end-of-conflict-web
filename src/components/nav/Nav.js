/**
 * EoC Nav
 * @flow
 */
import React, { useState } from 'react';
import { NavLink } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import './Nav.css';

type Props = {
  chapters: Array<{
    title: string,
    url: string
  }>,
  language: string
};

const Nav = (props: Props): React$Element<"div"> => {
  const { chapters, language } = props;
  const [open, setOpen] = useState(false);

  function openNav() {
    setOpen(!open);
  }

  return (
    <div className={`menu${open ? ' open' : ''}`}>
      <nav>
        <NavLink to="/">
          Home
        </NavLink>
        {
          chapters
            ? chapters.map((item, index) => {
                const { title, url } = item;
                return <NavLink to={`/chapters/${language}${url}`} key={index}>{title}</NavLink>
              })
            : null
        }
      </nav>
      <div className="toggle" onClick={openNav}>
        <FontAwesomeIcon icon={open ? 'times' : 'bars'} />
      </div>
    </div>
    
  );
}

Nav.defaultProps = {
  chapters: ([]: Array<empty>),
  language: 'en'
}

export default Nav;
