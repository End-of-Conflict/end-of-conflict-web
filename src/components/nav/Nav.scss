@import '../../base/breakpoints';

.menu {
  display: flex;
  padding: 1rem 0;
  justify-content: flex-end;

  @include breakpoint(sm) {
    align-items: center;
    justify-content: flex-end;
  }

  .toggle {
    cursor: pointer;
    position: absolute;
    top: .5rem;
    right: 1rem;

    @include breakpoint(sm) {
      position: relative;
      top: auto;
      right: auto;
    }
  }

  nav {
    display: none;

    @include breakpoint(sm) {
      align-items: center;
    }

    a {
      margin-right: .5rem;

      &:last-child {
        margin-right: 0;

        @include breakpoint(sm) {
          margin-right: .5rem;
        }
      }
    }
  }

  &.open {
    nav {
      display: flex;
    }
  }
}
