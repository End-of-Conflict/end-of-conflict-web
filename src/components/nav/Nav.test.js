import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { createSerializer } from 'enzyme-to-json';
import Nav from './Nav';
expect.addSnapshotSerializer(createSerializer({mode: 'deep'}));
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual for all non-hook parts
  useRouteMatch: () => ({
    url: '/en/0',
    params: { chapter: 0 }
  }),
}));
configure({ adapter: new Adapter() });

describe('Renders', () => {
  test('Nav', () => {
    const wrapper = shallow(<Nav />);
    expect(wrapper).toMatchSnapshot();
  });
});
