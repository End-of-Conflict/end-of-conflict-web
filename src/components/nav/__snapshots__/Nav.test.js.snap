// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Nav 1`] = `
<div
  className="menu"
>
  <div
    className="toggle"
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="bars"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      swapOpacity={false}
      symbol={false}
      title=""
      transform={null}
    />
  </div>
  <nav>
    <NavLink
      to="/"
    >
      Home
    </NavLink>
  </nav>
</div>
`;
